### PR TITLE
Add Central Management feature

### PR DESCRIPTION
**Pull request by exekias**
Kibana will offer Central Management for Beats, this is the workflow to use it:

- Enroll the Beat into Kibana, using the new added `enroll` command:

```
metricbeat enroll http://localhost:5601 d16a2ede945b4d9395c1ce0ff6e663f1
```

- Once running, the beat will retrieve configs from Kibana and apply the changes locally.

All state is stored under `meta/management.yml`

Note: this is a merge from a feature branch, all PRs to it have been reviewed already

### CODE REVIEW

**Comment by exekias** [comment]
Note to self: this PR uses `satori/go.uuid` lib, which is being replaced in #8458, we can expect some minor conflicts depending on which one goes in first

File: libbeat/kibana/client.go [hunk 1]
```diff
@@ -145,8 +145,10 @@ func NewClientWithConfig(config *ClientConfig) (*Client, error) {
 		},
 	}
 
-	if err = client.SetVersion(); err != nil {
-		return nil, fmt.Errorf("fail to get the Kibana version: %v", err)
+	if !config.IgnoreVersion {
```
Comment (ruflin) [inline libbeat/kibana/client.go]: Can you remind me in which case we need this?

Comment (exekias) [reply libbeat/kibana/client.go]: basically when doing enrollment, Kibana will probably be using x-pack security, so you need user/password to access the version, we don't have it, as enrollment endpoints don't require username/password but a valid token.

This change allows to configure the client to ignore kibana version check when doing enrollment

**Comment by exekias** [comment]
Yes, there is an ongoing effort for docs here: https://github.com/elastic/beats/issues/8470

File: libbeat/kibana/client.go (hunk 1 above)
Comment (ruflin) [reply libbeat/kibana/client.go]: Got it. Chicken / egg problem here :-)

File: libbeat/publisher/pipeline/pipeline.go [hunk 2]
```diff
@@ -151,6 +153,7 @@ type queueFactory func(queue.Eventer) (queue.Queue, error)
 // queue and outputs will be closed.
 func New(
 	beat beat.Info,
+	monitors Monitors,
```
Comment (urso) [inline libbeat/publisher/pipeline/pipeline.go]: Todo: monitors contains logger, metrics + telemetry registry. Why still pass 'metrics', is this a different instance?

File: libbeat/monitoring/report/elasticsearch/elasticsearch.go [hunk 3]
```diff
@@ -172,6 +172,7 @@ func makeReporter(beat beat.Info, settings report.Settings, cfg *common.Config)
 
 	pipeline, err := pipeline.New(
 		beat,
+		pipeline.Monitors{},
```
Comment (urso) [inline libbeat/monitoring/report/elasticsearch/elasticsearch.go]: Why pass an empty instance here? This removes the `x-pack.monitoring.output` metrics from each beat.

It's also a good idea to create a logger named `x-pack.monitoring`.

**Pull request merged**